### PR TITLE
Make valid grades shared constant

### PR DESCRIPTION
**Pull request by dmcavoy**
We were dynamically generating the list of valid grades for a section and then passing it down to the front end. Instead we should use a shared constant. 

## Testing story

- Updated and added some unit tests
- Relying on existing tests
- Tested setting grade on section

Screen Shot 2022-03-07 at 3 44 29 PM

### CODE REVIEW

File: lib/cdo/shared_constants.rb [hunk 1]
```diff
@@ -50,6 +50,23 @@ module SharedConstants
...
+    5
+    6
+    7
+    8
+    9
+    10
+    11
+    12
+    Other
+  ).freeze
```
Comment (davidsbailey) [inline lib/cdo/shared_constants.rb]: super tiny nit:
```suggestion
  STUDENT_GRADE_LEVELS = %w(K 1 2 3 4 5 6 7 8 9 10 11 12 Other).freeze
```

**Review by davidsbailey — APPROVED** [review]
Looks great @dmcavoy ! Thanks for doing this!

File: apps/src/templates/teacherDashboard/OwnedSectionsTable.jsx [hunk 2]
```diff
@@ -162,7 +159,7 @@ class OwnedSectionsTable extends Component {
     if (this.state.sortingColumns[gradeCol]) {
       const mult = directionArray[0] === 'asc' ? 1 : -1;
       return sortBy(data, function(obj) {
-        return mult * GRADES.indexOf(obj.grade);
+        return mult * StudentGradeLevels.concat(null).indexOf(obj.grade);
```
Comment (jamescodeorg) [inline apps/src/templates/teacherDashboard/OwnedSectionsTable.jsx:162]: For my education, why is the null at the end of the array needed?

**Review by jamescodeorg — APPROVED** [review]
🎉

File: apps/src/templates/teacherDashboard/OwnedSectionsTable.jsx (hunk 2 above)
Comment (dmcavoy) [reply apps/src/templates/teacherDashboard/OwnedSectionsTable.jsx:162]: I think because grade can be empty. It matched the existing set up of GRADES

**Pull request merged**